### PR TITLE
Bugfix: Allow opensearch.bat file and opensearch-env.bat files to run when install path includes a space.

### PR DESCRIPTION
**Pull request by mwilkinson-imo**
### Description
This change enables the opensearch.bat and opensearch-env.bat files to run successfully when the install path includes a space. Example: "C:\Program Files (x86)\OpenSearch\bin"

### Issues Resolved
Resolves an undocumented bug related to Windows installation -https://github.com/opensearch-project/opensearch-build/issues/33

Before:
image

After:

image

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Hi @mwilkinson-imo thanks for the contribution.
Is it possible to move the `"` for other env vars so they will not have this issue later on?

Thanks.

**Comment by mwilkinson-imo** [comment]
> Hi @mwilkinson-imo thanks for the contribution. Is it possible to move the `"` for other env vars so they will not have this issue later on?
> 
> Thanks.

Just finished applying it to the other variables in that block and re-validating my install. Can run a few other install kinds to validate I haven't broken anything.

File: distribution/src/bin/opensearch-env.bat [hunk 1]
```diff
@@ -43,14 +43,14 @@ rem comparing to empty string makes this equivalent to bash -v check on env var
 rem and allows to effectively force use of the bundled jdk when launching OpenSearch
 rem by setting OPENSEARCH_JAVA_HOME= and JAVA_HOME= 
 if not "%OPENSEARCH_JAVA_HOME%" == "" (
-  set JAVA="%OPENSEARCH_JAVA_HOME%\bin\java.exe"
+  set "JAVA=%OPENSEARCH_JAVA_HOME%\bin\java.exe"
```
[the remark this answers is not in the document]
Comment (mwilkinson-imo) [reply distribution/src/bin/opensearch-env.bat:46]: It has to do with how Batch sets environment variables.  
https://ss64.com/nt/set.html 

set JAVA="my/path" 
echo %JAVA% 
 -> Prints "my/path"

Including the variable name in the assignment excludes the double-quotes.
set "JAVA=my/path" 
echo %JAVA%
 -> prints my/path

**Comment by mwilkinson-imo** [comment]
Confirmed that install works in base case (no spaces in path) and new case (spaces in path).

File: distribution/src/bin/opensearch-env.bat (hunk 1 above)
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply distribution/src/bin/opensearch-env.bat:46]: @mwilkinson-imo ^^

[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply distribution/src/bin/opensearch-env.bat:46]: What I find:
```
The characters <, >, |, &, and ^ are special command shell characters, and they must be preceded by the escape character (^) or enclosed in quotation marks when used in <string> (for example, "StringContaining&Symbol"). If you use quotation marks to enclose a string that contains one of the special characters, the quotation marks are set as part of the environment variable value.
```

https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/set_1

**Comment by peterzhuamazon** [comment]
@reta @dblock please review I am good with this change.
Thanks @mwilkinson-imo.

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
@dblock LGTM, checked on my Windows box, seems to be working just fine, thanks @mwilkinson-imo

**Comment by dblock** [comment]
Thanks for fixing this @mwilkinson-imo !

If you want to do something else for Windows here, https://github.com/opensearch-project/OpenSearch/issues/1448 could use help.

**Comment by dblock** [comment]
@mwilkinson-imo do a manual PR into the 2.x branch so we can ship it with the next 2.x version, auto-backport failed ^

**Comment by peterzhuamazon** [comment]
> @mwilkinson-imo do a manual PR into the 2.x branch so we can ship it with the next 2.x version, auto-backport failed ^

I think 1.3/1.x/2.3 branch is also needed here.
Thanks.

**Comment by dblock** [comment]
How much do we want to backport windows fixes into 1.x?

**Comment by reta** [comment]
It probably won't hurt to have it in 1.x (but it is low risk change, why not)?

**Comment by peterzhuamazon** [comment]
I will backport to all the branches mentioned above.

**Comment by peterzhuamazon** [comment]
* https://github.com/opensearch-project/OpenSearch/pull/4470
* https://github.com/opensearch-project/OpenSearch/pull/4472
* https://github.com/opensearch-project/OpenSearch/pull/4473

**Pull request merged**